ibm/resource_ibm_network_gateway_test.go

```diff
@@ -0,0 +1,152 @@
+
+	hostname := acctest.RandString(16)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMNetworkGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:  testAccCheckIBMNetworkGatewayConfig_basic(hostname),
+				Destroy: false,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMNetworkGatewayExists("ibm_network_gateway.terraform-acceptance-test-1", &networkGateway),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "hostname", hostname),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "domain", "terraformuat.ibm.com"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "os_reference_code", "OS_VYATTA_5600_5_X_UP_TO_1GBPS_SUBSCRIPTION_EDITION_64_BIT"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "datacenter", "ams01"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "network_speed", "100"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "hourly_billing", "false"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "private_network_only", "false"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "user_metadata", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "tcp_monitoring", "true"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "process_key_name", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "os_key_name", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "redundant_network", "false"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "public_bandwidth", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "memory", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "ipv6_enabled", "true"),
+					CheckStringSet(
+						"ibm_network_gateway.terraform-acceptance-test-1",
+						"tags", []string{"collectd"},
+					),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMNetworkGatewayConfig_basic(hostname string) string {
+	return fmt.Sprintf(`
+resource "ibm_network_gateway" "terraform-acceptance-test-1" {
+			hostname               = "%s"
+			domain                 = "terraformuat.ibm.com"
+			os_reference_code      = "OS_VYATTA_5600_5_X_UP_TO_1GBPS_SUBSCRIPTION_EDITION_64_BIT"
```

Comment:
`os_reference_code` is not needed in the schema. `os_key_name` will do the job.

Reply:
os_reference_code has been removed